**4.3.11** Assuming that $495$ divides $273x49y5$, obtain the digits $x$ and $y$.

I see that $495$ and $273x49y5$ (let’s call that $N$) are both divisible by $5$; what else divides $495$?

In [1]:
using Primes
factor(495)

3^2 ⋅ 5 ⋅ 11

A-ha! $495$ is also divisible by $9$ ($3^2$) and by $11$. I know things about divisibility by $9$ and $11$.

Since $9 \mid 495$ and $495 \mid N$, then $9 \mid N$. Thanks to our theorem, we know that $9$ also divides the sum of the digits of $N$, that is:

$$9 \mid 2+7+3+x+4+9+y+5$$

Letting Julia do the arithmetic:

In [2]:
2 + 7 + 3 + 4 + 9 + 5

30

shows that $9$ must divide $x + y + 30$.

I wonder which combinations of $x$ and $y$ satisfy this?

In [3]:
𝐴 = [(𝑥 = 𝑥, 𝑦 = 𝑦) for 𝑦 ∈ 0:9, 𝑥 ∈ 0:9]
𝐵 = filter(𝑡 -> (𝑡.𝑥 + 𝑡.𝑦 + 30) % 9 == 0, 𝐴)

11-element Array{NamedTuple{(:𝑥, :𝑦),Tuple{Int64,Int64}},1}:
 (𝑥 = 0, 𝑦 = 6)
 (𝑥 = 1, 𝑦 = 5)
 (𝑥 = 2, 𝑦 = 4)
 (𝑥 = 3, 𝑦 = 3)
 (𝑥 = 4, 𝑦 = 2)
 (𝑥 = 5, 𝑦 = 1)
 (𝑥 = 6, 𝑦 = 0)
 (𝑥 = 6, 𝑦 = 9)
 (𝑥 = 7, 𝑦 = 8)
 (𝑥 = 8, 𝑦 = 7)
 (𝑥 = 9, 𝑦 = 6)

In the code above, I make a list ($A$) of all the combinations of digits $x$ and $y$; then, I filter that list down to the ones where $9 \mid x + y + 30$. There were eleven of them. (*The filter function in Julia takes a function that returns true or false and list of items. It applies the function to each item of the list and returns a new list containing the items that the function returned true for.*)

There is a further restriction on $x$ and $y$ due to the divisibility of $495$ and $N$ by $11$. It must also be the case that $11$ divides the alternating sum of the digits of $N$. That is:

$$11 \mid 5-y+9-4+x-3+7-2$$

More arithmetic:

In [4]:
5 + 9 - 4 - 3 + 7 - 2

12

So, our second restriction on $x$ & $y$ is that $11$ must divide $x-y+12$. The combinations of $x$ and $y$ that satisfy this are:

In [5]:
𝐶 = filter(𝑡 -> (𝑡.𝑥 - 𝑡.𝑦 + 12) % 11 == 0, 𝐴)

9-element Array{NamedTuple{(:𝑥, :𝑦),Tuple{Int64,Int64}},1}:
 (𝑥 = 0, 𝑦 = 1)
 (𝑥 = 1, 𝑦 = 2)
 (𝑥 = 2, 𝑦 = 3)
 (𝑥 = 3, 𝑦 = 4)
 (𝑥 = 4, 𝑦 = 5)
 (𝑥 = 5, 𝑦 = 6)
 (𝑥 = 6, 𝑦 = 7)
 (𝑥 = 7, 𝑦 = 8)
 (𝑥 = 8, 𝑦 = 9)

I could probably eyeball the two lists to find the combination that satisfies both conditions, but just in case:

In [6]:
𝐵 ∩ 𝐶

1-element Array{NamedTuple{(:𝑥, :𝑦),Tuple{Int64,Int64}},1}:
 (𝑥 = 7, 𝑦 = 8)

This should mean that $495 \mid 27374985$. These are small enough to check directly:

In [7]:
divrem(27374985, 495)

(55303, 0)

That is, $27374985 = 55303 \times 495 + 0$. I’ll just mention here that $273x49y5$ is divisible by $5$ for all values of digits $x$ and $y$.

Again, since these numbers are so small, we could have applied what we know about decimal representation to find $N$ more directly:

In [8]:
filter(𝑡 -> (27300000 + 𝑡.𝑥 * 10000 + 4900 + 𝑡.𝑦 * 10 + 5) % 495 == 0, 𝐴)

1-element Array{NamedTuple{(:𝑥, :𝑦),Tuple{Int64,Int64}},1}:
 (𝑥 = 7, 𝑦 = 8)

Happily, this gives the same value for $N$: $27374985$.

---

Another note about Julia: it is new enough that Unicode was already well-established when it was created. As long as there are not conflicts, just about any unicode characters can be used for variables. Interestingly, this means that the characters in Unicode’s [Mathematical Alphanumeric Symbols](https://en.wikipedia.org/wiki/Mathematical_Alphanumeric_Symbols) block are prefectly valid variable names, letting us use the same symbols we would traditionally use to typeset math with. The biggest challenge is making sure that whatever is displaying the code has access to fonts with the needed glyph repertoire. 